pkg/reconciler/sequence/sequence.go

```diff
@@ -190,7 +192,7 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, step int, p *v1.
 		// TODO: Send events here, or elsewhere?
 		//r.Recorder.Eventf(p, corev1.EventTypeWarning, subscriptionCreateFailed, "Create Sequences's subscription failed: %v", err)
 		return nil, fmt.Errorf("failed to get subscription: %s", err)
-	} else if !equality.Semantic.DeepDerivative(expected.Spec, sub.Spec) {
+	} else if equal, err := kmp.SafeEqual(sub.Spec, expected.Spec, ignoreArguments); !equal || err != nil {
```

Comment:
As only .spec.channel is relevant here, we could also compare only these fields (instead of using `ignoreArguments`). So this might be a bit more self speaking directly :shrug:

Reply:
I just had it from the comment `Given that spec.channel is immutable...`. Also in the `SubscriptionSpec` this is the only one marked as immutable: https://github.com/knative/eventing/blob/746dc52c1a2fb3b152dad71d62e60c09d8a3fa9f/pkg/apis/messaging/v1/subscription_types.go#L74-L107

But aligning with the webhook code (via function call) makes sense to me